### PR TITLE
Add head_with_check_avs func to solve #543

### PR DESCRIPTION
**Pull request by mocchira**
#543 and #546 will be solved by using head_with_check_avs/2

### CODE REVIEW

**Comment by yosukehara** [comment]
LINK: leo-project/leofs/issues/543

File: src/leo_object_storage_haystack.erl [hunk 1]
```diff
@@ -305,6 +305,51 @@ head_with_calc_md5(MetaDBId, StorageInfo, Key, MD5Context) ->
...
+        {ok, MetadataBin} ->
+            Metadata_1 = binary_to_term(MetadataBin),
+            #backend_info{read_handler = ReadHandler} = StorageInfo,
+            HeaderSize = erlang:round(?BLEN_HEADER/8),
+            case leo_file:pread(ReadHandler, Metadata_1#?METADATA.offset, HeaderSize) of
+                {ok, HeaderBin} ->
+                    case leo_object_storage_transformer:header_bin_to_metadata(HeaderBin) of
+                        {error, _Cause} ->
+                            {error, _Cause};
+                        _Metadata ->
```
Comment (yosukehara) [inline src/leo_object_storage_haystack.erl]: @mocchira I thought it would be better to compare `Metadata_1` with `_Metadata`.

Comment (mocchira) [reply src/leo_object_storage_haystack.erl]: @yosukehara I think the complete check (check the all fields one by one) is a bit overspec and consume a bit cpu cycle more than necessary (as header_bin_to_metadata already checks if each filed is valid or not), how about checking only a few fields in #?METADATA like
- addr_id
- ksize

?

Comment (yosukehara) [reply src/leo_object_storage_haystack.erl]: @mocchira I've understood. [This row](https://github.com/leo-project/leo_object_storage/pull/20/files#diff-5f19f9835375c9b5354f0f744b008963R326) checks the Metadata's format. It is enough to us. This PR will be merged soon.

**Pull request merged**